specs/experimental/gov-proposal-validator.md

```diff
@@ -0,0 +1,581 @@
+- `bool isRelativeToVotableSupply`: True if the against threshold should be relative to the votable supply.
+
+`submitCouncilMemberElectionsProposal`
+
+Submits a Council Member Elections proposal for approval and voting.
+
+- MUST be called by an approved address
+- MUST check if the proposal is a duplicate
+- MUST use the `Approval` Voting Module
+- MUST use "TopChoices" criteria type for the Voting Module
+- MUST provide a valid attestation UID
+- MUST NOT execute any operations
+- MUST emit `ProposalSubmitted` and `ProposalVotingModuleData` events
+- MUST store submission proposal data which are defined by the `ProposalSubmissionData` struct
+
+```solidity
+function submitCouncilMemberElectionsProposal(
+    uint128 _criteriaValue,
+    string[] _optionDescriptions,
+    string memory _proposalDescription,
+    bytes32 _attestationUid
+) external returns (bytes32 proposalHash_);
+```
+
+Approval Voting Module
+
+Council Member Elections proposals use the `Approval` voting module.
+This requires the user who submits the proposal to provide some additional data related to the proposal.
+
+For the `ProposalSettings` of the voting module, these are:
+- `uint128 criteriaValue`: Since the passing criteria type is "TopChoices" this number represents the amount
+of top choices that can pass the voting.
+
+For the `ProposalOptions` of the voting module, these are:
+- `string[] optionDescriptions`: The strings of the different options that can be voted.
+
+`submitFundingProposal`
+
+Submits a `GovernanceFund` or `CouncilBudget` proposal type that transfers OP tokens for approval and voting.
+
+- MUST only be called for `GovernanceFund` or `CouncilBudget` proposal type
+- CAN be called by anyone
+- MUST check if the proposal is a duplicate
+- MUST use the `Approval` Voting Module
+- MUST use "Threshold" criteria type for the Voting Module
+- MUST use the `Predeploys.GOVERNANCE_TOKEN` and `IERC20.transfer` signature to create the `calldata`
+for each option
+- MUST NOT request to transfer more than `proposalDistributionThreshold` tokens
+- MUST emit `ProposalSubmitted` event
+- MUST store submission proposal data which are defined by the `ProposalSubmissionData` struct
+
+```solidity
+function submitFundingProposal(
+    uint128 _criteriaValue,
+    string[] _optionsDescriptions,
+    address[] _optionsRecipients,
+    uint256[] _optionsAmounts,
+    string memory _description,
+    ProposalType _proposalType,
+) external returns (bytes32 proposalHash_);
+```
+
+Approval Voting Module
+
+Funding proposals use the `Approval` voting module but unlike the Protocol or Governor upgrade proposals,
+funding proposals need to execute token transfers.
+This requires the user who submits the proposal to provide some additional data related to the proposal.
+
+For the `ProposalSettings` of the voting module, these are:
+- `uint128 criteriaValue`: Since the passing criteria type is always "Threshold" for this proposal type
```

Comment:
I believe that this should be a fixed % of votable supply.  So I think we can remove it as an argument and just pass fixed % of the votable supply oracle value